### ISSUE TITLE
Protect user profiles with ML security

### ISSUE THREAD

**Issue by grtjn** [marklogic/slush-marklogic-node#241]
proxy.js currently contains special measure to safeguard against profile updates on other users profiles. I think it will make things easier if we simply disallow updating user profiles (with protected collections or such), and force the ui to use an amped rest extension to do profile updates.

**Comment by janmichaelyu** [comment]
@grtjn , by amped rest extension do you mean a new rest extension needs to be created and the profile route will only use that new extension?

**Comment by grtjn** [comment]
Yes. Some profile rest extension which is amped itself or uses an amped function to access docs in uris/collections otherwise restricted..

I think we used to have something like that in demo-cat at some point..

**Comment by janmichaelyu** [comment]
PR #334
